src/components/v2/appDetails/sourceInfo/EnvironmentSelector.component.tsx

```diff
@@ -0,0 +1,142 @@
+import React, { useEffect, useState } from 'react';
+import Select, { components } from 'react-select';
+import { multiSelectStyles, SingleSelectOption as Option } from '../../../common';
+import ScalePodModalComponent from './ScalePodModal.component';
+import './sourceInfo.css';
+import IndexStore from '../index.store';
+import { AppEnvironment } from './environment.type';
+import { useParams, useHistory, useRouteMatch, generatePath, Route, useLocation } from 'react-router';
+
+import { getAppOtherEnvironment } from '../appDetails.api';
+import { useSharedState } from '../../utils/useSharedState';
+
+function EnvironmentSelectorComponent() {
+    const params = useParams<{ appId: string; envId?: string }>();
+    const { url, path } = useRouteMatch();
+    const history = useHistory();
+
+    const [showhiberbateConfirmationModal, setshowHibernateConfirmationModal] = useState(false);
+
+    const [environments, setEnvironments] = useState<Array<AppEnvironment>>();
+
+    const [appDetails] = useSharedState(IndexStore.getAppDetails(), IndexStore.getAppDetailsObservable());
+
+    useEffect(() => {
+        if (params.appId) {
+            getAppOtherEnvironment(params.appId)
```

Comment:
in case of error, it may be because application is not deployed to any environment, in this case we need to have handling like we have for http://localhost:3000/app/317/details

Reply:
that will not happen in case of charts,,, for the case of apps we will consider it later